utils/google-drive.js

```diff
@@ -59,104 +82,205 @@ const updateMetadata = ({metadata, fieldsDefault = {}, fieldsMapper = {}}) => {
   return {...metadata, breadcrumb}
 }
 
-async function fetchTree({
-  debug,
-  breadcrumb,
-  folderId,
-  fields,
-  ignoredFolders = [],
-}) {
+async function getGdrive() {
   const googleOAuth2 = new GoogleOAuth2({
     token: ENV_TOKEN_VAR,
   })
   const auth = await googleOAuth2.getAuth()
 
-  return new Promise((resolve, reject) => {
-    google.drive({version: "v3", auth}).files.list(
-      {
-        includeTeamDriveItems: true,
-        supportsAllDrives: true,
-        q: `${
-          folderId ? `'${folderId}' in parents and ` : ""
-        }(mimeType='${MIME_TYPE_FOLDER}' or mimeType='${MIME_TYPE_DOCUMENT}') and trashed = false`,
-        fields: `files(id, mimeType, name, description, createdTime, modifiedTime, starred${
-          fields ? `, ${fields.join(", ")}` : ""
-        })`,
-      },
-      async (err, res) => {
-        if (err) {
-          return reject(err)
-        }
-
-        const documents = res.data.files.filter(
-          (file) => file.mimeType === MIME_TYPE_DOCUMENT
-        )
-        const rawFolders = res.data.files.filter(
-          (file) => file.mimeType === MIME_TYPE_FOLDER
-        )
+  return google.drive({version: "v3", auth})
+}
 
-        let folders = []
-        for (const folder of rawFolders) {
-          if (
-            folder.name.toLowerCase() === "drafts" ||
-            ignoredFolders.includes(folder.name) ||
-            ignoredFolders.includes(folder.id)
-          ) {
-            continue
-          }
-
-          if (debug) {
-            const breadCrumbString =
-              breadcrumb.length > 0 ? breadcrumb.join("/") + "/" : ""
-            // eslint-disable-next-line no-console
-            console.info(
-              `source-google-docs: Fetching ${breadCrumbString}${folder.name}`
-            )
-          }
-
-          const files = await fetchTree({
+/**
+ * @typedef DocumentFetchParent
+ * @property {string | null} id
+ * @property {string[]} breadcrumb
+ * @property {string} path
+ */
+
+/**
+ * @typedef FetchDocumentsOptions
+ * @property {import('googleapis').drive_v3.Drive} drive
+ * @property {DocumentFetchParent[]} parents
+ */
+
+// 10 per 1.5 seconds.
+const rateLimit = wyt(10, 1500)
```

Comment:
Did you find an official documentation about this or did you lower until it works?